### PR TITLE
Add FileSystem API to read/write files and use it

### PR DESCRIPTION
**Pull request by McGiverGim**
This adds a FileSystem module as wraper for file functions. In this way I think it will be easier to migrate to Capacitor if needed in a future.

Includes, at this moment, the firmware flasher tab. Others are in progress...

### CODE REVIEW

**Comment by McGiverGim** [comment]
Rebased against master and updated the CLI code.

File: src/js/FileSystem.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        });
+
+        const file = this._createFile(fileHandle);
+
+        if (await this.verifyPermission(file, true)) {
+            return file;
+        }
+    }
+
+    async pickReadFile(description, extension) {
```
Comment (haslinghuis) [inline src/js/FileSystem.js:29]: Suggest using Open instead of Read

```suggestion
    async pickOpenFile(description, extension) {
```

**Review by haslinghuis — APPROVED** [review]
Code looks good. One suggestion.

File: src/js/FileSystem.js (hunk 1 above)
Comment (McGiverGim) [reply src/js/FileSystem.js:29]: Ok, I will change...

**Comment by McGiverGim** [comment]
Modified as per review. Added VTX tab.

**Comment by McGiverGim** [comment]
Added OSD tab

**Comment by McGiverGim** [comment]
Almost there... only missing is the Blackbox, download to a file... it's different than the others, because it writes by chunks...

**Comment by McGiverGim** [comment]
I think it's ready for review :)

File: src/js/LogoManager.js [hunk 2]
```diff
@@ -229,23 +229,26 @@ LogoManager.hideUploadHint = function () {
...
-            // load and validate selected image
-            const img = new Image();
-            img.onload = () => {
-                validateImage.apply(this, [img])
-                    .then(() => resolveOpenImage(img))
-                    .catch(error => rejectOpenImage(error));
-            };
-            img.onerror = error => rejectOpenImage(error);
-            fileEntry.file(file => img.src = `file://${file.path}`);
+        //FileSystem.pickOpenFile(i18n.getMessage('fileSystemPickerFiles', {typeof: this.acceptFileTypes[0].description.toUpperCase()}), `.${this.acceptFileTypes[0].extensions.join('.,')}`)
```
Comment (haslinghuis) [inline src/js/LogoManager.js:232]: leftover

File: src/js/tabs/onboard_logging.js [hunk 3]
```diff
@@ -489,7 +492,13 @@ onboard_logging.initialize = function (callback) {
 
                     const startTime = new Date().getTime();
                     // Fetch the initial block
-                    mspHelper.dataflashRead(nextAddress, self.blockSize, onChunkRead);
+                    FileSystem.openFile(fileWriter).
+                    then((file) => {
+                        openedFile = file;
+                        mspHelper.dataflashRead(nextAddress, self.blockSize, onChunkRead);
+                    }).then(() => {
+                        //return FileSystem.closeFile(openedFile);
```
Comment (haslinghuis) [inline src/js/tabs/onboard_logging.js:500]: Can be removed

File: src/js/LogoManager.js (hunk 2 above)
Comment (McGiverGim) [reply src/js/LogoManager.js:232]: Fixed

File: src/js/tabs/onboard_logging.js (hunk 3 above)
Comment (McGiverGim) [reply src/js/tabs/onboard_logging.js]: Fixed

File: src/js/tabs/vtx.js [hunk 4]
```diff
@@ -640,44 +640,21 @@ vtx.initialize = function (callback) {
...
+            FileSystem.writeFile(file, text);
 
-            }, function (){
-                console.error('Failed to get VTX table lua file writer');
-                gui_log(i18n.getMessage('vtxSavedLuaFileKo'));
-            });
+            tracking.sendEvent(tracking.EVENT_CATEGORIES.FLIGHT_CONTROLLER, 'VtxTableLuaSave', { length: text.length });
+        })
+        .catch((error) => {
+            console.error("FAiled to write lua file:", error);
```
[the remark this answers is not in the document]
Comment (McGiverGim) [reply src/js/tabs/vtx.js]: Fixed

**Comment by nerdCopter** [comment]
`27d764d5` local `yarn dev`:
- flashed from cloud-build & flashed from local hex.
- uploaded font with custom logo.
- pseudo-setup unsoldered FC.
- loaded VTX table via json file.
- CLI `Save to file` (`diff all`).
- applied same file via CLI `Load from file`.

**Review by nerdCopter — APPROVED** [review]

**Pull request merged**